### DISCUSSION TITLE
[Auth] ReCaptcha Flow and restricted keys

### DISCUSSION THREAD

**Discussion by KDederichs** [invertase/react-native-firebase#5452] [Q&A] [answered]
Hey,
I'm running into an issue (mainly on emulators) where Phone Auth will redirect to the ReCaptcha Web flow.
The problem here is that it uses the iOS and Android specific keys which I'd like to restrict to the application itself for obvious reasons.
Since those keys are bound to the app the web recaptcha flow will fail though, since it's not executing them through the app but to through a browser and as a result I'll get a 403 - Requests from this application blocked error.

Is there a solution to this except removing the application restrictions from the keys?

**Comment by mikehardy** [comment, marked as answer]
Hi there! I'm not sure exactly, this seems like a more general firebase usage question which is best posed either upstream in firebase-android-sdk / firebase-ios-sdk or on stackoverflow with appropriate tags (the firebase folks offer authoritative answers there)

**Reply by KDederichs** [reply]
Thanks, I'll go as in upstream then!
